oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2AuthorizationCodeRequestAuthenticationProvider.java

```diff
@@ -109,6 +111,12 @@ public Authentication authenticate(Authentication authentication) throws Authent
 					authorizationCodeRequestAuthentication, null);
 		}
 
+		if (registeredClient.getClientSecretExpiresAt() != null &&
```

Comment:
@ansatsing This check has already been implemented in gh-850

Reply:
I know the checkpoint you mentioned. But the earlier the system checks, the less meaningless processing it will do.